sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -351,6 +351,11 @@ static int _set(gnrc_netif_t *netif, const gnrc_netapi_opt_t *opt)
     switch (opt->opt) {
         case NETOPT_LORAWAN_DR:
             assert(opt->data_len == sizeof(uint8_t));
+            if (*((uint8_t *)opt->data) & 0xF0) {
```

Comment:
it's not valid to assume that DR0 - DR7 are all valid.
For instance, DR7 is not supported right now in RIOT. 

For now, I would change it to DR0 - DR5 (DR6 is not added to GNRC LoRaWAN, although we can easily add it).

Reply:
I am a bit confused here. DR would always be 0-15 hench check here makes sense. Validating if the DR is valid in current system environment is not a thing of `netif` right?

Reply:
>  am a bit confused here. DR would always be 0-15

E.g in Europe, DR > 7 doesn't exist (Please refer to the to the LoRaWAN regional specification: https://lora-alliance.org/wp-content/uploads/2020/11/lorawan_regional_parameters_v1.0.3reva_0.pdf). Therefore this is not true.

> Validating if the DR is valid in current system environment should not be of concern to netif right?

Of course it is. Otherwise this PR wouldn't make any sense.
What is not a scope of the netif is the implementation details regarding _why_ a datarate is valid or not (this is also region specific). Therefore, there's a public API to leverage that to GNRC LoRaWAN (`gnrc_lorawan_validate_dr`)